src/Evolution/Systems/Cce/Components/WorldtubeBoundary.hpp

```diff
@@ -75,4 +75,72 @@ struct H5WorldtubeBoundary {
     }
   }
 };
+
+/*!
+ * \brief Component that supplies CCE worldtube boundary data sourced from a
+ * running GH system.
+ *
+ * \details The \ref DataBoxGroup associated with the worldtube boundary
+ * component contains an interface manager (derived from
+ * `Cce::GHWorldtubeInterfaceManager`) that stores and provides the data
+ * received from the GH system (via the simple action
+ * `Cce::Actions::ReceiveGHWorldtubeData`). The data manager handles buffering
+ * and interpolating to desired target time points when requested via the simple
+ * action `Cce::Actions::BoundaryComputeAndSendToEvolution`, at which point it
+ * will send the required collection of boundary quantities to the identified
+ * `CharacteristicEvolution` component. It is assumed that the simple action
+ * `Cce::Actions::BoundaryComputeAndSendToEvolution` will only be called during
+ * the `Evolve` phase.
+ *
+ * Uses const global tags:
+ * - `InitializationTags::LMax`
+ * - `InitializationTags::ExtractionRadius`
+ *
+ * `Metavariables` must contain:
+ * - the `enum` `Phase` with at least `Initialization` and `Evolve` phases.
+ * - a type alias `cce_boundary_communication_tags` for the set of tags to send
+ * from the worldtube to the evolution component. This will typically be
+ * `Cce::Tags::characteristic_worldtube_boundary_tags<Tags::BoundaryValue>`.
+ */
+template <class Metavariables>
+struct GHWorldtubeBoundary {
+  using chare_type = Parallel::Algorithms::Singleton;
+  using metavariables = Metavariables;
+  using initialize_action_list =
+      tmpl::list<Actions::InitializeGHWorldtubeBoundary,
+                 Initialization::Actions::RemoveOptionsAndTerminatePhase>;
+  using initialization_tags =
+      Parallel::get_initialization_tags<initialize_action_list>;
+
+  using worldtube_boundary_computation_steps = tmpl::list<>;
+
+  using phase_dependent_action_list =
+      tmpl::list<Parallel::PhaseActions<typename Metavariables::Phase,
+                                        Metavariables::Phase::Initialization,
+                                        initialize_action_list>,
+                 Parallel::PhaseActions<typename Metavariables::Phase,
+                                        Metavariables::Phase::Evolve,
+                                        worldtube_boundary_computation_steps>>;
+
+  using const_global_cache_tag_list =
+      Parallel::detail::get_const_global_cache_tags_from_pdal<
+          phase_dependent_action_list>;
+
+  using options = tmpl::list<>;
+
+  static void initialize(Parallel::CProxy_ConstGlobalCache<
+                         Metavariables>& /*global_cache*/) noexcept {}
+
+  static void execute_next_phase(
+      const typename Metavariables::Phase next_phase,
+      const Parallel::CProxy_ConstGlobalCache<Metavariables>&
+          global_cache) noexcept {
+    auto& local_cache = *(global_cache.ckLocalBranch());
+    if (next_phase == Metavariables::Phase::Evolve) {
```

Comment:
do you need this call?

Reply:
and if you need it, you should call `start_phase`

Reply:
ping an calling `start_phase` here instead of `perform_algorithm`. I'm surprised it works at all with `perform_algorithm` because it's `start_phase` that actually updates the phase on the component's end.